jsonschema-core.xml

```diff
@@ -800,49 +800,302 @@
+                    SHOULD NOT serve a document at that URL.  A server MAY respond with
+                    the relevant protocol's successful "no content" message, such as
+                    an HTTP 204 status.
+                    <cref>
+                        Vocabulary documents may be added shortly, or in the next draft.
+                        For now, identifying the keyword set is deemed sufficient as that,
+                        along with meta-schema validation, is how the current "vocabularies"
+                        work today.
+                    </cref>
+                </t>
+                <t>
+                    The values of the object properties MUST be booleans.
+                    If the value is true, then implementations that do not recognize
+                    the vocabulary MUST refuse to process any schemas that declare
+                    this meta-schema with "$schema".  If the value is false, implementations
+                    that do not recognize the vocabulary MAY choose to proceed with processing
+                    such schemas.
+                </t>
+                <t>
+                    When processing a schema that uses unrecognized vocabularies, keywords
+                    declared by those vocabularies are treated like any other unrecognized
+                    keyword, and ignored.
+                </t>
+                <t>
+                    The "$vocabulary" keyword SHOULD be used in the root schema of any schema
+                    document intended for use as a meta-schema.  It MUST NOT appear in subschemas.
+                </t>
+                <t>
+                    The "$vocabulary" keyword MUST be ignored in schema documents that
+                    are not being processed as a meta-schema.  This allows validating
+                    a meta-schema M against its own meta-schema M' without requiring
+                    the validator to understand the vocabularies declared by M.
+                </t>
+                <t>
+                    Note that the processing restrictions on "$vocabulary" mean that
+                    meta-schemas that reference other meta-schemas using "$ref" or
+                    similar keywords do not automatically inherit the vocabulary
+                    declarations of those other meta-schemas.  All such declarations
+                    must be repeated in the root of each schema document intended
+                    for use as a meta-schema.  This is demonstrated in
+                    <xref target="example-meta-schema">the example meta-schema</xref>.
+                </t>
+                <t>
+                    If "$vocabulary" is absent, an implementation MAY determine
+                    behavior based on the meta-schema if it is recognized from the
+                    URI value of the referring schema's "$schema" keyword.
+                    If the meta-schema, as referenced by the schema, is not recognized,
```

Comment:
Doesn't this clash a bit with what is stated in line 1028?

Reply:
Hmm...  not exactly but I see why it looks that way, and there is probably a missing piece.

If you can read a meta-schema (because you have a copy of it) and there is no `$vocabulary`, line 1028 is saying that you MUST assume the use of core.  This part here on line 939-940 is saying that you MAY assume additional vocabularies to produce identical behavior to the pre-`$vocabulary` word.

But we should probably put all of this guidance together, because it's easy to read 1028 as also implying that since you MUST assume the core vocabulary, you MUST NOT assume other vocabularies.  Which is probably not quite right.

Reply:
It's very clear now.